packages/flutter_tools/test/commands.shard/hermetic/precache_test.dart

```diff
@@ -41,8 +33,7 @@ void main() {
     );
     await createTestCommandRunner(command).run(const <String>['precache']);
 
-    // Do not throw StateError, lock is acquired.
-    expect(() => cache.checkLockAcquired(), returnsNormally);
+    expect(cache.locked, true);
```

Comment:
I'm just curious, does this command not release the lock at the end?

Reply:
It does - this test is just asserting that the command locked it at the start

Reply:
Does `await`-ing `runner.run()` not block until the command exits?

Reply:
the command never exits in the test scripts

Reply:
TIL

Reply:
ahh, actually no - the command acquires the lock but the release is managed by the runner elsewhere during shutdown or Cache.releaseLockEarly